presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +373,17 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+
+    @Test
+    public void testPruneCountAggregationOverScalar()
```

Comment:
@kokosing could you please point me to where and when in the code this rewrite happens? I got as far as RelationPlanner and QueryPlanner but I dont see where an EXISTS is getting rewritten to an AggregationNode. I'm assuming there is some AstVisitor that does this...